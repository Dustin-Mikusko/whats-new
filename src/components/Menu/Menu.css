.menu {
  padding: 20px 0px 0px 20px;
  color: white;
}

.new {
  color: rgb(255, 183, 0);
}

h1 {
  margin-bottom: 60px;
}

h2 {
  text-align: left;
  margin-bottom: 20px;
  font-weight: 400;
  height: 70px;
  padding: 20px 0px 0px 10px;
}

.topic {
  margin-left: 30px;
  width: 225px
}

.topic:hover {
  cursor:pointer;
  transform: scale(1.07);
  background-color: rgb(255, 183, 0);
  border-radius: 0px 50px 50px 0px;
}

.topic:focus {
  background-color: rgb(255, 183, 0);
  border-radius: 0px 50px 50px 0px;
}
