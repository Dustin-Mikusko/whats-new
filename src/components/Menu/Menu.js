import React from 'react';
import './Menu.css'

const Menu = () => {
  return (
    <header className="menu">
      <h1>What's <span className="new">New?</span></h1>
      <h2 className="topic">Local News</h2>
      <h2 className="topic">Technology</h2>
      <h2 className="topic">Entertainment</h2>
      <h2 className="topic">Science</h2>
      <h2 className="topic">Health</h2>
    </header>
  )
}

export default Menu;
