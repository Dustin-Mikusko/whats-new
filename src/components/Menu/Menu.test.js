import React from 'react';
import Menu from './Menu';
import { shallow } from 'enzyme';

describe('Menu', () => {
  let wrapper;
  let changeNewsMock;

  beforeEach(() => {
    changeNewsMock = jest.fn();
    wrapper = shallow(<Menu
      changeNews={changeNewsMock}
    />);
  })

  it('should match the snapshot with all the data passed in correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should call highlightSelection and changeNews when submitSelection is called ', () => {
    const mockEvent = { target: { dataset: { id: 'local' } } };
    
    wrapper.instance().highlightSelection = jest.fn();

    wrapper.instance().submitSelection(mockEvent);

    expect(changeNewsMock).toHaveBeenCalledWith('local');
  })

})
