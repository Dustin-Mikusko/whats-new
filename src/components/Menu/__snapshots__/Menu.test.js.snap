// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu should match the snapshot with all the data passed in correctly 1`] = `
<header
  className="menu"
>
  <h1>
    What's 
    <span
      className="new"
    >
      New?
    </span>
  </h1>
  <h2
    className="topic active"
    data-id="local"
    onClick={[Function]}
  >
    <GoLocation />
     Local News
  </h2>
  <h2
    className="topic"
    data-id="technology"
    onClick={[Function]}
  >
    <GiGears />
     Technology
  </h2>
  <h2
    className="topic"
    data-id="entertainment"
    onClick={[Function]}
  >
    <GiTicket />
     Entertainment
  </h2>
  <h2
    className="topic"
    data-id="science"
    onClick={[Function]}
  >
    <GiMicroscope />
     Science
  </h2>
  <h2
    className="topic"
    data-id="health"
    onClick={[Function]}
  >
    <FaHeartbeat />
     Health
  </h2>
</header>
`;
